packages/hierarchies/learning/HierarchyLevelFiltering.md

```diff
@@ -0,0 +1,137 @@
+# Hierarchy level filtering
+
+Hierarchy level filtering is a concept where filtering is applied to a single hierarchy level (as opposed to [hierarchy filtering](./HierarchyFiltering.md), where filtering is applied to the whole hierarchy). This is useful when hierarchy levels contain very large numbers of nodes as it allows users to filter the list based on attributes of instances represented by the nodes.
+
+## Enabling hierarchy level filtering
+
+The library creates hierarchies as described by hierarchy definitions and they define hierarchy levels through ECSQL queries. Due to loose structure of the queries and the need to be flexible on how the filter is applied on them, the library does not provide a built-in mechanism for hierarchy level filtering. Instead, it provides a way to enable consumers to implement filtering through their hierarchy definitions.
+
+### Enable filtering for specific parent nodes
+
+Because hierarchy definitions may not support hierarchy level filtering on all hierarchy levels (and by default they don't support it any of them), the hierarchy definition has to explicitly tell if node supports it or not.
```

Comment:
```suggestion
Because hierarchy definitions may not support hierarchy level filtering on all hierarchy levels (and by default they don't support it on any of them), the hierarchy definition has to explicitly tell if node supports it or not.
```